azure-pipelines.yml

```diff
@@ -25,13 +25,13 @@ jobs:
      - template: install-rust.yml@templates
        parameters:
          rust: $(rust)
-     - script: cargo check --all-targets
+     - script: cargo check --all-targets --all-features
        displayName: cargo check
-     - script: cargo test --examples
+     - script: cargo test --examples --all-features
        displayName: cargo test --examples
-     - script: cargo test --doc
+     - script: cargo test --doc --all-features
        displayName: cargo test --doc
-     - script: cargo test --lib
+     - script: cargo test --lib --all-features
```

Comment:
Now that we have more features, I'd like us to start using [`cargo hack`](https://github.com/taiki-e/cargo-hack) to test all combinations of features. There's an example job template for it [over in inferno's CI](https://github.com/jonhoo/inferno/blob/43ed7801cdea55af186c054fe8280c6ca426bff3/azure-pipelines.yml#L11-L22).

Reply:
Oh, this is nice. I added this to the azure pipeline. The `cargo hack` line works for me locally, if it blows up in Azure then I'll figure it out.